src/wallet/feebumper.cpp

```diff
@@ -293,21 +295,35 @@ Result CreateRateBumpTransaction(CWallet& wallet, const uint256& txid, const CCo
     // We cannot source new unconfirmed inputs(bip125 rule 2)
     new_coin_control.m_min_depth = 1;
 
-    constexpr int RANDOM_CHANGE_POSITION = -1;
-    auto res = CreateTransaction(wallet, recipients, RANDOM_CHANGE_POSITION, new_coin_control, false);
-    if (!res) {
-        errors.push_back(Untranslated("Unable to create transaction.") + Untranslated(" ") + util::ErrorString(res));
-        return Result::WALLET_ERROR;
-    }
+    if (recipients.size() > 0) {
+        constexpr int RANDOM_CHANGE_POSITION = -1;
+        auto res = CreateTransaction(wallet, recipients, RANDOM_CHANGE_POSITION, new_coin_control, false);
+        if (!res) {
+            errors.push_back(Untranslated("Unable to create transaction.") + Untranslated(" ") + util::ErrorString(res));
+            return Result::WALLET_ERROR;
+        }
 
-    const auto& txr = *res;
-    // Write back new fee if successful
-    new_fee = txr.fee;
+        const auto& txr = *res;
+        // Write back new fee if successful
+        new_fee = txr.fee;
 
-    // Write back transaction
-    mtx = CMutableTransaction(*txr.tx);
+        // Write back transaction
+        mtx = CMutableTransaction(*txr.tx);
 
-    return Result::OK;
+        return Result::OK;
+    } else {
+        // The transaction only has one output, and it is either detected as change,
+        // or selected as the one to take the fee from.
+        // In that case, just reduce its value to meet the new feerate.
+        new_fee = new_coin_control.m_feerate->GetFee(maxTxSize);
+        temp_mtx.vout[0].nValue = input_value - new_fee;
```

Comment:
There still is a problem with this approach when the user sets a value for the single output.

Basically, the user provided value is being discarded. We don't check if it's within the inputs amount bounds nor if need to create a change output for the remaining amount (value can be lower than the inputs amount).

Current code instead of create a change output when the value is lower than the inputs amount, it sets the value equal to inputs total minus new fee. We should code a test for it too.

Thus why thought about using SFFO in 27195 to mark the output from where we want to subtract fees, so the apparent "single output tx" can be crafted using `CreateTransaction` instead of making an special case here.

Side from that, all good about closing 27195. I made it just because it's a bug in master.

Reply:
Ah, I see. Actually I've decided to just rebase this PR on top of #27195.

Reply:
k, either way is fine for me. Cool about the special case exception removal and the new test.
just updated 27195 with further tests too.